### PR TITLE
Move Config Guides for Pre-Built Detection Rules to Setup Field - Windows, MacOS, BBR and Cross Platform

### PR DESCRIPTION
**Pull request by shashank-elastic**
## Issues
https://github.com/elastic/ia-trade-team/issues/149

## Summary

Based on the Analysis of detection rules, the following rules had an existing Setup Guide, they have been aligned and updated to setup filed available in 8.3+

- cross-platform - All checks done. 
- macOS - All checks are done. 
- Windows - All checks are done. 
- Linux - All checks are done. 

At this point, no Setup Guide needs to be aligned and all Setup Guides have the setup filed alignment 

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/cross-platform/discovery_security_software_grep.toml [hunk 1]
```diff
@@ -50,7 +50,12 @@ This rule looks for the execution of the `grep` utility with arguments compatibl
 
 ## Setup
 
-If enabling an EQL rule on a non-elastic-agent index (such as beats) for versions <8.2, events will not define `event.ingested` and default fallback for EQL rules was not added until 8.2, so you will need to add a custom pipeline to populate `event.ingested` to @timestamp for this rule to work.
+If enabling an EQL rule on a non-elastic-agent index (such as beats) for versions <8.2,
+events will not define `event.ingested` and default fallback for EQL rules was not added until version 8.2.
+Hence for this rule to work effectively, users will need to add a custom injest pipeline to populate
+`event.ingested` to @timestamp.
+For more details to add a custom injest pipeline refer - https://www.elastic.co/guide/en/fleet/current/data-streams-pipeline-tutorial.html
```
Comment (Aegrah) [inline rules/cross-platform/discovery_security_software_grep.toml:57]:
```suggestion
For more details on adding a custom ingest pipeline refer - https://www.elastic.co/guide/en/fleet/current/data-streams-pipeline-tutorial.html
```

File: rules/cross-platform/discovery_security_software_grep.toml [hunk 2]
```diff
@@ -50,7 +50,12 @@ This rule looks for the execution of the `grep` utility with arguments compatibl
 
 ## Setup
 
-If enabling an EQL rule on a non-elastic-agent index (such as beats) for versions <8.2, events will not define `event.ingested` and default fallback for EQL rules was not added until 8.2, so you will need to add a custom pipeline to populate `event.ingested` to @timestamp for this rule to work.
+If enabling an EQL rule on a non-elastic-agent index (such as beats) for versions <8.2,
+events will not define `event.ingested` and default fallback for EQL rules was not added until version 8.2.
+Hence for this rule to work effectively, users will need to add a custom injest pipeline to populate
```
Comment (Aegrah) [inline rules/cross-platform/discovery_security_software_grep.toml:55]:
```suggestion
Hence for this rule to work effectively, users will need to add a custom ingest pipeline to populate
```

**Review by Aegrah — COMMENTED** [review]
A few nits. These should be applied to the other setup guides as well!

**Comment by Mikaayenson** [comment]
Before merging this in, please work with DED to open a maintenance window, so we can address the backports.

cc. @eric-forte-elastic @terrancedejesus

**Review by imays11 — APPROVED** [review]

**Review by approksiu — COMMENTED** [review]
Please mention all the integrations that user can use to get the data for the rule. Provide links to the documentation for those, and mention if any specific integration policies need to be applied. Thanks!

**Comment by shashank-elastic** [comment]
Linux Rules New Config information of Elastic Defend And Auditbeat
- https://github.com/elastic/detection-rules/pull/3177

**Review by DefSecSentinel — APPROVED** [review]
LGTM

**Pull request merged**